### PR TITLE
Fix lamini-examples for 2.5.6 release (also maintain backward compatibility)

### PR DESCRIPTION
**Pull request by yx-lamini**
Tested with the following process, for lamini 2.5.6, 2.5.2:
```
cd 02_eval
./scripts/eval.sh
cd ../03_prompt_tuning
./scripts/prompt_tune.sh
cd ../04_rag_tuning
./scripts/rag_tuning.sh
```
Eyeballed the output, seems fine.

### CODE REVIEW

File: 03_prompt_tuning/spot_check.py [hunk 1]
```diff
@@ -109,13 +98,28 @@ def get_company_info(example):
     return prompt
 
 
+class LaminiModelStage(GenerationNode):
+    def __init__(self, model_name="meta-llama/Meta-Llama-3-8B-Instruct",):
+        super().__init__(
+            model_name=model_name,
+            max_new_tokens=150,
+        )
+        self.model_name = model_name
```
Comment (powerjohnnyli) [inline 03_prompt_tuning/spot_check.py:107]: don't need self.model_name here

we can remove this __init__ and just do the model_name, max_new_tokens init at the instantiation like this
 
self.model_stage = LaminiModelStage(
            "meta-llama/Meta-Llama-3-8B-Instruct", max_new_tokens=150
        )

File: 02_eval/eval.py [hunk 2]
```diff
@@ -74,8 +73,45 @@ def load_dataset(args):
...
+        **kwargs,
+    ):
+        results = super().generate(
+            prompt,
+            output_type=self.dataset.get_output_type(),
+            *args,
+            **kwargs,
+        )
+
+        return results
```
Comment (powerjohnnyli) [inline 02_eval/eval.py:98]: We can remove the __init__ and generate() functions

for generate we can pass output_type in the pipeline like this
```
def forward(self, x):
        x = self.question_generator(
            x,
            output_type=dataset.get_output_type()
        )
        x = self.answer_generator(x)
        return x
        
      ```
      
      See below for how to change __init__

**Review by powerjohnnyli — APPROVED** [review]

File: 03_prompt_tuning/spot_check.py (hunk 1 above)
Comment (yx-lamini) [reply 03_prompt_tuning/spot_check.py:107]: Done.

File: 02_eval/eval.py (hunk 2 above)
Comment (yx-lamini) [reply 02_eval/eval.py]: Done
Moved the LaminiModelStage to eval_pipeline.py to be close to other subclasses of GenerationNode
Also rewrote ModifyStage ScoreStage to use its designed pattern

**Comment by yx-lamini** [comment]
https://github.com/lamini-ai/lamini-examples/actions/runs/10070658262
Passed

**Pull request merged**